.upstream-tests/test/std/atomics/atomics.types.generic/floating_point.pass.cpp

```diff
@@ -0,0 +1,193 @@
+//     bool compare_exchange_strong(floating_point& expc, floating_point desr,
+//                                  memory_order s, memory_order f);
+//     bool compare_exchange_weak(floating_point& expc, floating_point desr,
+//                                memory_order m = memory_order_seq_cst) volatile;
+//     bool compare_exchange_weak(floating_point& expc, floating_point desr,
+//                                memory_order m = memory_order_seq_cst);
+//     bool compare_exchange_strong(floating_point& expc, floating_point desr,
+//                                 memory_order m = memory_order_seq_cst) volatile;
+//     bool compare_exchange_strong(floating_point& expc, floating_point desr,
+//                                  memory_order m = memory_order_seq_cst);
+//
+//     floating_point
+//         fetch_add(floating_point op, memory_order m = memory_order_seq_cst) volatile;
+//     floating_point fetch_add(floating_point op, memory_order m = memory_order_seq_cst);
+//     floating_point
+//         fetch_sub(floating_point op, memory_order m = memory_order_seq_cst) volatile;
+//     floating_point fetch_sub(floating_point op, memory_order m = memory_order_seq_cst);
+//     floating_point
+//
+//     atomic() = default;
+//     constexpr atomic(floating_point desr);
+//     atomic(const atomic&) = delete;
+//     atomic& operator=(const atomic&) = delete;
+//     atomic& operator=(const atomic&) volatile = delete;
+//     floating_point operator=(floating_point desr) volatile;
+//     floating_point operator=(floating_point desr);
+//
+//     floating_point operator++(int) volatile;
+//     floating_point operator++(int);
+//     floating_point operator--(int) volatile;
+//     floating_point operator--(int);
+//     floating_point operator++() volatile;
+//     floating_point operator++();
+//     floating_point operator--() volatile;
+//     floating_point operator--();
+//     floating_point operator+=(floating_point op) volatile;
+//     floating_point operator+=(floating_point op);
+//     floating_point operator-=(floating_point op) volatile;
+//     floating_point operator-=(floating_point op);
+// };
+
+#include <cuda/std/atomic>
+#include <cuda/std/cassert>
+
+#include <cmpxchg_loop.h>
+
+#include "test_macros.h"
+#if !defined(TEST_COMPILER_C1XX)
+  #include "placement_new.h"
+#endif
+#include "cuda_space_selector.h"
+
+template <class A, class T, template<typename, typename> class Selector>
+__host__ __device__ __noinline__
+void
+do_test()
+{
+    Selector<A, constructor_initializer> sel;
+    A & obj = *sel.construct(T(0));
+    bool b0 = obj.is_lock_free();
+    ((void)b0); // mark as unused
+    obj.store(T(0));
+    assert(obj == T(0));
+    obj.store(T(1), cuda::std::memory_order_release);
+    assert(obj == T(1));
+    assert(obj.load() == T(1));
+    assert(obj.load(cuda::std::memory_order_acquire) == T(1));
+    assert(obj.exchange(T(2)) == T(1));
```

Comment:
I know this is the libcxx style, but could we at least add some newlines between the different functionality that is tested.

Reading a gazillion consecutive lines drains a lot of brain power